File: Server/Models/Recipe.cs
using Microsoft.EntityFrameworkCore.ValueGeneration.Internal;

namespace Server.Models
{
    public class Recipe
    {
        public int Id { get; set; }
        public string Name { get; set; } = "";
        public string Description { get; set; } = "";
        public List<Ingredient> Ingredients { get; set; } = new();
    }

    public class Ingredient
    {
        public int Id { get; set; }
        public int RecipeId { get; set; }
        public string Name { get; set; } = "";
        public string Comment { get; set; } = "";
        public decimal Weight { get; set; }
        public decimal ProteinPer100 { get; set; }
        public decimal CaloriePer100 { get; set; }
        public int FdcId { get; set; }
    }
}

File: Client/src/models/Recipe.js
export class Recipe {
  constructor(id, name, description, ingredients = []) {
    this.id = id
    this.name = name
    this.description = description
    this.ingredients = ingredients
  }
}

export class Ingredient {
  constructor(id, recipeId, name, comment, weight, proteinPer100, caloriePer100, fdcId) {
    this.id = id
    this.recipeId = recipeId
    this.name = name
    this.comment = comment ?? ""
    this.weight = Number(weight)
    this.proteinPer100 = proteinPer100
    this.caloriesPer100 = caloriePer100
    this.fdcId = fdcId
  }
}
